src/string_array.c

```diff
@@ -87,6 +88,53 @@ rcutils_string_array_fini(rcutils_string_array_t * string_array)
   return RCUTILS_RET_OK;
 }
 
+rcutils_ret_t
+rcutils_string_array_cmp(
+  const rcutils_string_array_t * lhs,
+  const rcutils_string_array_t * rhs,
+  int * res)
+{
+  if (NULL == lhs || NULL == rhs) {
+    RCUTILS_SET_ERROR_MSG("string_array is null");
+    return RCUTILS_RET_INVALID_ARGUMENT;
+  }
+  if (NULL == lhs->data || NULL == rhs->data) {
+    RCUTILS_SET_ERROR_MSG("string_array data is null");
+    return RCUTILS_RET_INVALID_ARGUMENT;
+  }
+  if (NULL == res) {
+    RCUTILS_SET_ERROR_MSG("res argument is null");
+    return RCUTILS_RET_INVALID_ARGUMENT;
+  }
```

Comment:
@jacobperron for above checks, why not use `RCUTILS_CHECK_ARGUMENT_FOR_NULL()` and `RCUTILS_CHECK_FOR_NULL_WITH_MSG()` macros instead?

Reply:
I was looking at how the error handling was done in the other functions in this file. Updated this function in 5774096. We can update the others in a follow-up PR.